lib/experimental/class-wp-rest-navigation-fallback-controller.php

```diff
@@ -0,0 +1,161 @@
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		// Getting fallbacks requires creating and reading `wp_navigation` posts.
+		if ( ! current_user_can( $post_type->cap->create_posts ) || ! current_user_can( 'edit_theme_options' ) || ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_cannot_create',
+				__( 'Sorry, you are not allowed to create Navigation Menus as this user.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( 'edit' === $request['context'] && ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit Navigation Menus as this user.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Gets the most appropriate fallback Navigation Menu.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post = WP_Navigation_Fallback_Gutenberg::get_fallback();
+
+		if ( empty( $post ) ) {
+			return rest_ensure_response( new WP_Error( 'no_fallback_menu', __( 'No fallback menu found.', 'gutenberg' ), array( 'status' => 404 ) ) );
+		}
+
+		$response = $this->prepare_item_for_response( $post, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the fallbacks' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'navigation-fallbacks',
+			'type'       => 'object',
+			'properties' => array(
+				'id' => array(
+					'description' => __( 'The unique identifier for the Navigation Menu.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Matches the post data to the schema we want.
+	 *
+	 * @param WP_Post         $item The wp_navigation Post object whose response is being prepared.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response The response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data = array();
+
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = (int) $item->ID;
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
```

Comment:
@getdave What happened about the links?

Reply:
Links have now been added. I tried adding the following test but the `_links` field doesn't seem to be included even though in my manual testing of the endpoint it _is_ included. Any ideas?

```php
public function test_adds_links() {
	$request  = new WP_REST_Request( 'GET', '/wp-block-editor/v1/navigation-fallbacks' );
	$response = rest_get_server()->dispatch( $request );
	$data     = $response->get_data();

	$navigation_post_id = $data['id'];

	$this->assertArrayHasKey( '_links', $data, 'Response should contain links.' );

	$this->assertArrayHasKey( 'self', $data['_links'], 'Response should contain a "self" link.' );

	$this->assertStringContainsString( 'wp-json/wp/v2/navigation/' . $data['id'], $data['_links']['self'][0]['href'], 'Self link should reference the correct Navigation Menu post resource url.' );

	$this->assertTrue( $data['_links']['self'][0]['embeddable'], 'Self link should be embeddable.' );
}
```

Reply:
I added a test that uses https://developer.wordpress.org/reference/classes/wp_rest_response/get_links/ to assert on the content of `_links`.